erigon-lib/state/domain.go

```diff
@@ -1726,7 +1726,8 @@ func (dt *DomainRoTx) statelessBtree(i int) *BtIndex {
 
 func (dt *DomainRoTx) valsCursor(tx kv.Tx) (c kv.Cursor, err error) {
 	if dt.valsC != nil {
-		return dt.valsC, nil
+		dt.valsC.Close()
```

Comment:
this change looks not relevant

Reply:
This was a fix issued by @awskii , otherwise the history reader would return stale account data from time to time. So this is necessary for correctness.